### ISSUE TITLE
Backup Not fully Completing (Files Missing)

### ISSUE THREAD

**Issue by Tonkapug** [a4k-openproject/plugin.program.openwizard#102]
I'm running Kodi 18.7 on a Windows 10 PC

I have a large library integrated via TMDB Helper. When trying to backup using the Wizard it completes without error but the final product is missing numerous items. Including skin, settings, addons, databases potion of the library etc...

Here's what I did to test:
I renamed my build's directory (Kodi to Kodi_backup) and reopened Kodi to a fresh install. I installed only 'OpenWizard', TMDB Helper, and my skin.

I then completed the backup without the library included and everything is as it should be. No issues and the backup can be installed on any device I choose.

I then added the library (movies & tvshows directories) to TMDB directory in addon_data. I didn't bother adding the sources or scraping as this would take some time. I just wanted to see if the actually large library of .strm files was the cause. So, I completed the backup again and it produced the exact same result as my previous build. Numerous items are missing. Including skin, settings, addons, databases potion of the library etc... As soon as the library is added the backup fails.

The error I am seeing in the wizard log is "Files count would require ZIP64 extensions"

Here's what is contained in the backup.zip:

2 Directories - addons & userdata

Here's what is in the userdata directory:

addon_data (the skins directory with settings.xml is missing along with other addon data)
advancdsettings.xml
favorties.xml
guisettings.xml
profiles.xml
Rssfeeds.xml
sources.xml

I was speaking to SerpentDrago about this issue as well on the a4k sub_reddit
https://www.reddit.com/r/Addons4Kodi/comments/gsfpfa/openwizard_backup_missing_files/

I have attached 4 logs:

Backup without library (Completes as normal)
kodi-BACKUP_TEST1
[kodi-BACKUP_TEST1.log](https://github.com/drinfernoo/plugin.program.openwizard/files/4705424/kodi-BACKUP_TEST1.log)

wizard_BACKUP_TEST1
[wizard-BACKUP_TEST1.log](https://github.com/drinfernoo/plugin.program.openwizard/files/4705425/wizard-BACKUP_TEST1.log)

Backup with library (Completes but missing files)
kodi-BACKUP_TEST2
https://mega.nz/file/fCQFRSZa#8gLax8TTSvf4s4Mo8TMkiLAy0YAZNpqsuq1qm9j-VN4

wizard-BACKUP_TEST2
https://mega.nz/file/jfBzESpL#QKzKw3CkGNTndep88k-KiQqTk4JeEjAeniI3NW8oO_E

Hopefully, this can be resolved
Thanks for your help

**Comment by drinfernoo** [comment]
Hey, thanks for the detailed bug report! I actually think I should have it fixed in the latest `develop` branch, if you'd test it for me?

**Comment by Tonkapug** [comment]
Thank you for the quick response. I will test as soon as I return home and report back with the results.

Appreciate your work...enjoy your evening!

**Comment by Tonkapug** [comment]
I installed the development branch but received an error during install and also on the first run of the wizard. Below I have attached the Kodi log.

[kodi.log](https://github.com/drinfernoo/plugin.program.openwizard/files/4706718/kodi.log)

**Comment by Tonkapug** [comment]
Just checking in! Any luck sorting the issue? The dev branch will not run at all, it just produces an error and closes. Unable to test completing a backup.

Again, thanks for looking into this issue. Have a good evening!

**Comment by drinfernoo** [comment]
@Tonkapug Can you test it again? The latest `develop` should be working fine.

**Comment by Tonkapug** [comment]
I tested again and it installs fine. I tried on a couple of devices the other day and could not get it to install.

I completed a backup and it appears to have completed successfully. I checked the backup.zip and now the complete library is included along with the missing addon_data and databases.

Although, I attempted to restore the backup via 'Local Restore' and it fails with an error right at the start. I have attached both logs.

[kodi.log](https://github.com/drinfernoo/plugin.program.openwizard/files/4710069/kodi.log)

[wizard.log](https://github.com/drinfernoo/plugin.program.openwizard/files/4710071/wizard.log)

I also attempted via the builds menu after editing builds.txt with a URL and I get the following error after the zip is downloaded.

[2020-06-01 05:36:06] Error Checking Zip: Bad magic number for central directory

**Comment by Tonkapug** [comment]
Thanks for the help but I couldn't get the backup to restore using the development branch. I tried multiple times on multiple devices.  Googled the error and what I can only assume is the ZIP64 format is producing the error "Bad magic number error with ZipFile".

**Comment by drinfernoo** [comment]
Yeah... I need to figure out a better way. The wizard currently uses a modified version of the standard `zipfile` Python module, but I think to fix this may require use of the `libvfs.archive` module instead.

**Comment by Tonkapug** [comment]
If you figure it out just drop me a line and I'll test for you. I saved the library and will keep it until a fix is available.  I'm flying out to work in the AM and won't return for about a month, so this is all on the backburner till then anyway. As always, I appreciate your help and the wizard. Keep up the good work!
